crates/stages/src/stages/sender_recovery.rs

```diff
@@ -101,18 +101,19 @@ impl<DB: Database> Stage<DB> for SenderRecoveryStage {
         for chunk_range in chunks {
             // An _unordered_ channel to receive results from a rayon job
             let (recovered_senders_tx, recovered_senders_rx) = mpsc::channel();
-            channels.push(recovered_senders_rx);
+            channels.push((chunk_range.clone(), recovered_senders_rx));
 
             let static_file_provider = provider.static_file_provider().clone();
 
             // Spawn the task onto the global rayon pool
             // This task will send the results through the channel after it has read the transaction
             // and calculated the sender.
             rayon::spawn(move || {
+                debug!(target: "sync::stages::sender_recovery", ?chunk_range, "Recovering senders batch");
                 let mut rlp_buf = Vec::with_capacity(128);
                 let _ = static_file_provider.fetch_range_with_predicate(
                     StaticFileSegment::Transactions,
-                    chunk_range,
+                    chunk_range.clone(),
```

Comment:
can we not clone?, there are quite a few chunks

Reply:
I think that's fine

```
5 million commit threshold / 100 txs per chunk * (64 bits for range start + 64 bits for range end) = 800 kilobytes
```

even if we clone it twice (as we do now), it's just 1.6 megabytes